src/python/pants/backend/go/build.py

```diff
@@ -156,32 +145,27 @@ async def build_target(
     if request.is_main:
         import_path = "main"
 
-    argv = [
-        "./go/bin/go",
-        "tool",
-        "compile",
-        "-p",
-        import_path,
-        "-importcfg",
-        "./importcfg",
-        "-pack",
-        "-o",
-        "__pkg__.a",
-        *source_files.files,
-    ]
-
-    process = Process(
-        argv=argv,
-        env={
-            "GOROOT": "./go",
-        },
-        input_digest=input_digest,
-        output_files=["__pkg__.a"],
-        description=f"Compile Go package with {pluralize(len(source_files.files), 'file')}.",
-        level=LogLevel.DEBUG,
+    result = await Get(
+        ProcessResult,
+        SetupGoSdkProcess(
```

Comment:
I recommend simply `GoSdkProcess` - I think it reads a little confusing to have `await Get(ProcessResult, SetupGoSdkProcess)` because the setup part happens behind the scenes.

And then you'd have uniformity with Python land.

But not blocking.

Reply:
Done.